grid2op/multi_agent/multiAgentEnv.py

```diff
@@ -252,8 +261,23 @@ def step(self, action : ActionProfile) -> Tuple[MADict, MADict, MADict, MADict]:
             _description_
         """
         
-        # TODO
-        raise NotImplementedError()
+        order = self.agent_order
+        self._build_global_action(action, order)
+
+        self._cent_observation, reward, done, info = self._cent_env.step(self.global_action)
+        
+        self._dispatch_reward_done_info(reward, done, info)
+
+        self._update_observations()
+
+        return self.observations, self.rewards, self.done, self.info 
+    
+    def _dispatch_reward_done_info(self, reward, done, info):
+        for agent in self.agents:
+            self.rewards[agent] = reward
+            self._cumulative_rewards[agent] += reward
```

Comment:
we never mentionned that... And by the way it might not be accurate in case of "discount factor". 

Did you write somewhere that the discount factor is 1. ? I don't remember seeing it. But again that will not be my problem that will be yours, maybe

Reply:
Just removed it. It is not important at this stage.